### ISSUE TITLE
Dashboard filters send filter value as string - invalid Postgres query

### ISSUE THREAD

**Issue by c2h5oh** [metabase/metabase#3816]
- Your browser and the version: (e.x. Chrome 52.1, Firefox 48.0, IE 10)
Chrome 54.0.2840.100

- Your operating system:
Linux

- Your databases: (e.x. MySQL, Postgres, MongoDB, …)
PostgreSQL 9.6

- Metabase version:
v0.20.3

- Metabase hosting environment:
Ubuntu 16.04

- Metabase internal database:
H2

- *Repeatable steps to reproduce the issue*
1. Create a question and add it to a dashboard
2. Add dashboard filter that filters by a numeric column (doesn't matter if it's a PK or category)
3. Enter filter value (default or user entry - doesn't matter)

Expected: updated, filtered values
Actual: `There was a problem displaying this chart.`
Error: `operator does not exist: bigint = character varying\n  Hint: No operator matches the given name and argument type(s). You might need to add explicit type casts.`

Foreign keys seem to be working fine.

Payload modification experimentation results:
this fails
```json
{
   "cardID":5,
   "parameters":[
      {
         "type":"id",
         "target":[
            "dimension",
            [
               "field-id",
               119
            ]
         ],
         "value":"2158"
      }
   ]
}
```

this works: 
```json
{
   "cardID":5,
   "parameters":[
      {
         "type":"number",
         "target":[
            "dimension",
            [
               "field-id",
               119
            ]
         ],
         "value":"2158"
      }
   ]
}
```

**Comment by LukeAbell** [comment]
I'm having this same issue. 

<img width="1920" alt="screenshot_1422" src="https://cloud.githubusercontent.com/assets/3343530/20846634/9aa04f78-b898-11e6-9aa9-0affdd95157d.png">
<img width="1920" alt="screenshot_1426" src="https://cloud.githubusercontent.com/assets/3343530/20846636/9aa274ba-b898-11e6-9539-48ed9d8ed0cb.png">
<img width="1920" alt="screenshot_1427" src="https://cloud.githubusercontent.com/assets/3343530/20846635/9aa15c4c-b898-11e6-916e-a2e9633be2f2.png">
<img width="1684" alt="screenshot_1428" src="https://cloud.githubusercontent.com/assets/3343530/20846633/9aa008ec-b898-11e6-854d-7181506fea19.png">

**Comment by djtarazona** [comment]
Same issue here with Postgres.  However it works with tables where we use a UUID (string) as the primary key.  Let's make sure to not break that functionality (don't assume ID filters are always integers).

**Comment by MixMe** [comment]
i have same problem :(
Metabase version:
v0.22.0

**Comment by MixMe** [comment]
Please fix it, filters very powerful tool ! :(

**Comment by MixMe** [comment]
@camsaul Can you fix please this problem, we use Metabase for all our project, and we need filtration by ID.

**Comment by camsaul** [comment]
@MixMe we'll address this at some point in the future, but if you need it fixed right away, PRs are always welcome!

**Comment by MixMe** [comment]
@camsaul i don't now Clojure Language :(

**Comment by c2h5oh** [comment]
This is also broken for foreign keys

**Comment by tlrobinson** [comment]
We should fix this ASAP. I think the backend should be able to tell if the parameter should be a string or number. For dimensions on structured queries we know the underlying field type, and on native queries we know the `template_tag` type.

**Comment by camsaul** [comment]
Fixed by #4325

**Comment by valtlfelipe** [comment]
I'm having this issue in metabase `0.28.1` and PostgreSQL `9.5.10`.

I have a filter for the User ID in my dashboard and it throws this error: `ERROR: operator does not exist: bigint = character varying↵  Hint: No operator matches the given name and argument type(s). You might need to add explicit type casts.`

This is the payload: 
```json
{"parameters":[{"type":"id","target":["dimension",["fk->",42,209]],"value":"532"}]}
```
